File: Backend/Backend/Models/Veiculo.cs
using System.ComponentModel.DataAnnotations;
using Backend.Enums;

namespace Backend.Models
{
    public class Veiculo
	{
        [Key]
        public int Codigo { get; set; }
        public string Placa { get; set; }
        public int Ano { get; set; }
        public string Cor { get; set; }
        public string Modelo { get; set; }
        public TipoVeiculoEnum TipoVeiculo { get; set; }

        public virtual Carro? Carro { get; set; }
        public virtual Caminhao? Caminhao { get; set; }
        public List<Revisao>? Revisoes { get; set; }
    }
}

File: Frontend/src/app/models/veiculo.ts
import { TipoVeiculoEnum } from "../enum/TipoVeiculoEnum"
import { Caminhao } from "./caminhao"
import { Carro } from "./carro"
import { Revisao } from "./revisao"

export class Veiculo{
    carro:Carro|null = null
    caminhao:Caminhao| null = null
    revisoes:Revisao[] = []
    
    constructor(
        public codigo:number, 
        public modelo:string, 
        public ano:number, 
        public cor:string, 
        public placa:string,
        public tipoVeiculo:TipoVeiculoEnum | null){
    }
}